nebullvm/utils/feedback_collector.py

```diff
@@ -0,0 +1,167 @@
+        message = (
+            "Would you like to contribute to nebullvm continuous improvement? "
+            "Press enter to share the performance achieved with nebullvm and "
+            '"No" to not contribute. You can find full details in the '
+            '"Sharing feedback to improve nebullvm" section of the '
+            'documentation. [enter/"No"] '
+        )
+        metadata = {}
+        flag = True
+        while flag:
+            try:
+                collect_feedback_bool = _input_with_timeout(message, timeout)
+            except TimeoutError:
+                collect_feedback_bool = "no"
+            if len(collect_feedback_bool) == 0:
+                flag = False
+                metadata["allow_feedback_collection"] = True
+            elif collect_feedback_bool.lower().strip() == "no":
+                flag = False
+                metadata["allow_feedback_collection"] = False
+            message = (
+                'Press enter to give your consent or type "No" to deny '
+                "consent."
+            )
+        return metadata
+
+    @staticmethod
+    def _generate_model_id(model_name: str):
+        return f"{str(uuid.uuid4())}_{hash(model_name)}"
+
+    def start_collection(self, model: Any, framework: DeepLearningFramework):
+        if not self.is_active:
+            return
+        if isinstance(model, str) or isinstance(model, Path):
+            model_name = str(model)
+        else:
+            model_name = model.__class__.__name__
+
+        self._model_id = self._generate_model_id(model_name)
+        self._model_info = {
+            "model_name": model_name,
+            "model_size": _read_model_size(model),
+            "framework": framework.value,
+        }
+        self._latency_dict = {}
+
+    def store_compiler_result(
+        self,
+        compiler: ModelCompiler,
+        q_type: Optional[QuantizationType],
+        perf_loss_ths: Optional[float],
+        latency: Optional[float],
+    ):
+        if self._model_id is None:
+            return
+        q_type_key = (
+            f"{q_type.value}_{perf_loss_ths}"
+            if q_type is not None and perf_loss_ths is not None
+            else "noopt"
+        )
+        compiler_dict = self._latency_dict.get(compiler.value, {})
+        compiler_dict[q_type_key] = latency if latency else -1.0
+        self._latency_dict[compiler.value] = compiler_dict
+
+    def send_feedback(self):
+        if self._model_id is None:
+            return {}
+        request_body = {
+            "model_id": self._model_id,
+            "latency_dict": self._latency_dict,
+            "hardware_setup": self._hw_info,
+            "model_metadata": self._model_info,
+            "nebullvm_version": VERSION,
+        }
+        headers = {
+            "accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        url = "https://nebuly.cloud/v1/store_nebullvm_results"
+        response = requests.post(
```

Comment:
What happens if the url is down? We should set the timeout argument, otherwise it will wait indefinitely

Reply:
solved in https://github.com/nebuly-ai/nebullvm/pull/74/commits/892c91bb8001698f6eb385215f68c3e7f91fff86